testing/scenariobuilder/todoerr_scenario.go

```diff
@@ -48,6 +51,21 @@ func (s *TodoErrScenario) ExpectLine(line string) *TodoErrScenario {
 	return s
 }
 
+// WithMetadata extends existing metadata with a multiple of key value pairs
+// expected within the `metadata` field of the TODOs
+func (s *TodoErrScenario) WithMetadata(metadata *map[string]string) *TodoErrScenario {
+	for k, v := range *metadata {
+		s.metadata[k] = v
+	}
+	return s
+}
+
+// WithMetadataEntry stores a single key-value pair that is expected in the `metadata` field
+func (s *TodoErrScenario) WithMetadataEntry(key string, value string) *TodoErrScenario {
```

Comment:
Same here in terms of naming

Reply:
`WithMetadataEntry` -> `WithJSONMetadataEntry`